frontend/src/utils.js

```diff
@@ -608,6 +608,43 @@ export const hasSatelliteInconsistency = (canteen) => {
   return canteen.satelliteCanteensCount !== canteen.satellites.length
 }
 
+export const lineMinistryRequired = (canteen, allSectors) => {
+  const concernedSectors = allSectors.filter((x) => !!x.hasLineMinistry).map((x) => x.id)
+  if (concernedSectors.length === 0) return false
+  return canteen.sectors.some((x) => concernedSectors.indexOf(x) > -1)
+}
+
+export const missingCanteenData = (canteen, store) => {
+  // TODO: what location data to we require at minimum?
+  const requiredFields = ["siret", "name", "cityInseeCode", "productionType", "managementType"]
+  const missingFieldLambda = (f) => !canteen[f]
+  const missingSharedRequiredData = requiredFields.some(missingFieldLambda)
+  if (missingSharedRequiredData) return true
+
+  // sectors checks
+  if (!canteen.sectors || !canteen.sectors.length) return true
+  if (lineMinistryRequired(canteen, store.state.sectors) && !canteen.lineMinistry) return true
+
+  // siret duplication check?
+
+  // production type specific checks
+  const yearlyMealCountKey = "yearlyMealCount"
+  const onSiteFields = ["dailyMealCount", yearlyMealCountKey]
+  const centralKitchenFields = [yearlyMealCountKey, "satelliteCanteensCount"]
+  const satelliteFields = ["centralProducerSiret"]
+
+  if (canteen.productionType === "central") {
+    return centralKitchenFields.some(missingFieldLambda)
+  } else if (canteen.productionType === "central_serving") {
+    return centralKitchenFields.some(missingFieldLambda) && onSiteFields.some(missingFieldLambda)
+  } else if (canteen.productionType === "site_cooked_elsewhere") {
+    return onSiteFields.some(missingFieldLambda) && satelliteFields.some(missingFieldLambda)
+  } else if (canteen.productionType === "site") {
+    return onSiteFields.some(missingFieldLambda)
+  }
+  return true // shouldn't get to here, indicates a bug in our logic/data
```

Comment:
the above is an if-else based on canteen productionType, and I was being paranoid that if we were to introduce a new production type, then it would be safer to not let them teledeclare so we don't forget to implement extra checks in the frontend